go/marshal/decode_test.go

```diff
@@ -192,6 +192,15 @@ func TestDecode(tt *testing.T) {
 	t(types.NewStruct("Abc", types.StructData{
 		"e": types.Bool(false),
 	}), &t4, aBc{false})
+
+	// Name of struct is irrelevant to unmarshalling structs.
```

Comment:
Can you add a test for where the struct name _does_ match and for when the destination struct is empty just for completeness?